src/main/java/mops/presentation/FileController.java

```diff
@@ -0,0 +1,55 @@
+package mops.presentation;
+
+import lombok.AllArgsConstructor;
+import mops.businesslogic.Account;
+import mops.businesslogic.FileService;
+import mops.businesslogic.utils.AccountUtil;
+import org.keycloak.adapters.springsecurity.token.KeycloakAuthenticationToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("material1/file")
+@AllArgsConstructor
+public class FileController {
+
+    /**
+     * Handles actions on single files: Delete, Preview, Download.
+     */
+    private FileService fileService;
+
+    /**
+     * @param token   a keycloak authentication token
+     * @param model   spring boot view model
+     * @param fileId the id of the requested file
+     * @return the route to template 'file'
+     */
+    @GetMapping("/{fileId}")
```

Comment:
Maybe we should think about two routes. One for downloading a file and one for showing it in the browser.

Reply:
the show template has info on the file (maybe a preview) and a download button (or follow link)

Reply:
Exactly - so the 
`@GetMapping("/{fileId}")`
opens a preview, with a seperate download button to another route?

Reply:
Yes exactly.

Reply:
what about the download route?